File: Easy/Java/459. Repeated Substring Pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        // Append the string twice, then drop the first and last letter.
        // See if the result still contains the original string s.
        String dropped = (s + s).substring(1, (s.length() * 2) - 1);
        return dropped.contains(s);
    }
}

File: Easy/JavaScript/459. Repeated Substring Pattern.js
var repeatedSubstringPattern = function(s) {
    // Append the string twice, then drop the first and last letter.
    // See if the result still includes the original string s.
    let dropped = (s + s).substring(1, (s.length * 2) - 1);
    return dropped.includes(s);
};
